docs/asyncapi-file.md

```diff
@@ -3,4 +3,105 @@ title: "AsyncAPI Specification File"
+  D(Render Engine) --> J(Java Spring Boot)
+```
+
+> :bulb: **Remember**
+> 
+> When you are writing your template, you get access to the specification file in two different forms; the originalAsyncApi which is a stringified version of the specification `.yml` file and the second is the asyncapiDocument.
+
+> :memo: **Note:** 
+> You can check out the structure of the asyncapiDocument [here](https://github.com/asyncapi/parser-js/blob/master/API.md#module_@asyncapi/parser+AsyncAPIDocument)
+
+In the section below, you will learn how to use either the **asyncapiString** or the **asyncapiDocument** in your template.
+#### Method 1: asyncapiString + template ##
+The first way to generate an message-based API template using the generator tool is by using the asyncapiString. The asynapiString is a stringified version of the specification YAML or JSON file you pass as a parameter to the CLI `ag` command. During the generation process, the generator library calls `generator.originalAsyncAPI` which returns a stringified version of the original spec file. This stringified version is what the generator library passes to the renderer engine and the renderer engine makes it accessible in your output alias the template.
+
+**An example of an asyncapiString:**
+```
+const asyncapiString = `
+asyncapi: 2.4.0
+info:
+  title: Example to show stringified specification file
+  version: 1.0.0
+...
+`;
+
+```
+The generator library generates templates using the `generator.generateFromString` instance method as shown in the code snippet below
+
+```
+generator
+  .generateFromString(asyncapiString)
+  .then(() => {
+    console.log('Done!');
+  })
+  .catch(console.error);
+  ```
+
+#### Method 2: asyncapiDocument+ template ##
+Once the specification YAML or JSON file is passed as an input to the generator library, it is passed on to the [Parser](parser.md) library which then manipulates the file to a more structured document called the asyncapiDocument. Once the parser returns the document to the generator, the generator passes it to the renderer engine. The renderer engine will make the asyncapiDocument accessible from your output alias the template.
+
+From the template, you can access the document's content, for example, to extract the version of the spec file used for your Readme, you can do that by running `asyncapiDocument.version()` or to get the message payload you use `asyncapiDocument.allMessages()` method.
+
+In the code snippet below, you'll see how you can access document contents of the asyncapiDocument in your template.
+
+```
+  const apiName = asyncapi.info().title();
+  const channels = asyncapi.channels();
+...
+```
+> :memo: **Note:**  
+> To learn about the various instances you can use to access the documents' content, please check [this file](https://github.com/asyncapi/parser-js/blob/master/API.md#module_@asyncapi/parser+AsyncAPIDocument) 
+
+## Advantage of using the asyncApiDocument instead of the asyncapiString in your template
```

Comment:
```suggestion

```

Honestly, this doesn't seem to merit a section. I would delete this header and move the below sentence to the beginning of the `#### Method 2: asyncapiDocument+ template` section.

Reply:
I agree ☝️